src/VisualStudio/Core/Def/Implementation/LanguageClient/LanguageServerClient.cs

```diff
@@ -2,81 +2,83 @@
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
-    internal abstract class AbstractLanguageServerClient : ILanguageClient
+    [ContentType(ContentTypeNames.CSharpContentType)]
+    [ContentType(ContentTypeNames.VisualBasicContentType)]
+    [Export(typeof(ILanguageClient))]
+    [ExportMetadata("Capabilities", "WorkspaceStreamingSymbolProvider")]
+    internal sealed class LanguageServerClient : ILanguageClient
     {
+        private const string ServiceHubClientName = "ManagedLanguage.IDE.LanguageServer";
+
         private readonly IThreadingContext _threadingContext;
         private readonly Workspace _workspace;
         private readonly IEnumerable<Lazy<IOptionPersister>> _lazyOptions;
         private readonly LanguageServerClientEventListener _eventListener;
         private readonly IAsynchronousOperationListener _asyncListener;
 
-        private readonly string _serviceHubClientName;
-        private readonly string _languageServerName;
-
         /// <summary>
         /// Gets the name of the language client (displayed to the user).
         /// </summary>
-        public abstract string Name { get; }
+        public string Name => ServicesVSResources.CSharp_VB_Language_Server_Client;
 
         /// <summary>
         /// Gets the configuration section names for the language client. This may be null if the language client
         /// does not provide settings.
         /// </summary>
-        public virtual IEnumerable<string> ConfigurationSections { get; } = null;
+        public IEnumerable<string> ConfigurationSections { get; } = null;
 
         /// <summary>
         /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
         /// This may be null if the client does not need custom initialization options.
         /// </summary>
-        public virtual object InitializationOptions { get; } = null;
+        public object InitializationOptions { get; } = null;
 
         /// <summary>
         /// Gets the list of file names to watch for changes.  Changes will be sent to the server via 'workspace/didChangeWatchedFiles'
         /// message.  The files to watch must be under the current active workspace.  The file names can be specified as a relative
         /// paths to the exact file, or as glob patterns following the standard in .gitignore see https://www.kernel.org/pub/software/scm/git/docs/gitignore.html files.
         /// </summary>
-        public virtual IEnumerable<string> FilesToWatch { get; } = null;
+        public IEnumerable<string> FilesToWatch { get; } = null;
 
 #pragma warning disable CS0067 // event never used - implementing interface ILanguageClient
         public event AsyncEventHandler<EventArgs> StartAsync;
         public event AsyncEventHandler<EventArgs> StopAsync;
 #pragma warning restore CS0067 // event never used
```

Comment:
💡 The warning suppression can be removed if this is updated to the expected form:

```csharp
event AsyncEventHandler<EventArgs> ILanguageClient.StartAsync { add { } remove { } }
event AsyncEventHandler<EventArgs> ILanguageClient.StopAsync { add { } remove { } }
```

Reply:
Ah interesting, hadn't seen that form before.  thanks!
Found another issue trying this out - StartAsync is actually used.